src/pudl/output/ferc1.py

```diff
@@ -871,3 +873,417 @@ def add_mean_cap_additions(steam_df):
+        ).xbrl_metadata[
+            [
+                "xbrl_factoid",
+                "calculations",
+                "xbrl_factoid_name_original",
+                "inter_table_calc_flag",
+            ]
+        ]
+        tbl = (
+            table_df.assign(table_name=table_name)
+            .rename(columns={xbrl_factoid_name: "xbrl_factoid"})
+            .merge(tbl_meta, how="left", on="xbrl_factoid", validate="m:1")
+        )
+        explosion_tables.append(tbl)
+
+    other_dimensions = get_other_dimensions(tbl_names=tables_to_explode.keys())
+    pks = get_exploded_pks(tbl_names=tables_to_explode.keys())
+    value_col = get_value_col(tbl_names=tables_to_explode.keys())
+    # REMOVE THE DUPLICATION
+    logger.info("CONCAT!")
+    exploded = (
+        pd.concat(explosion_tables)
+        .pipe(
+            remove_factoids_from_mutliple_tables,
+            tables_to_explode,
+            root_table,
+            value_col,
+            pks,
+        )
+        .pipe(remove_totals_from_other_dimensions, other_dimensions)
+        .pipe(remove_inter_table_calc_duplication)
+        .pipe(remove_intra_table_calculated_values)
+    )
+    return exploded
+
+
+def get_other_dimensions(tbl_names: list[str]) -> list[str]:
+    """Get all of the column names for the other dimensions."""
+    other_dimensions = []
+    for table_name in tbl_names:
+        other_dimensions.append(
+            pudl.transform.ferc1.FERC1_TFR_CLASSES[
+                table_name
+            ]().params.reconcile_table_calculations.subtotal_column
+        )
+    other_dimensions = [sub for sub in other_dimensions if sub]
+    return other_dimensions
+
+
+def get_exploded_pks(tbl_names: list[str]) -> list[str]:
+    """Get the joint primary keys of the exploded tables."""
+    pks = []
+    for table_name in tbl_names:
+        xbrl_factoid_name = pudl.transform.ferc1.FERC1_TFR_CLASSES[
+            table_name
+        ]().params.xbrl_factoid_name
+        pks.append(
+            [
+                col
+                for col in pudl.metadata.classes.Resource.from_id(
+                    table_name
+                ).schema.primary_key
+                if col != xbrl_factoid_name
+            ]
+        )
+    pks = pudl.helpers.dedupe_n_flatten_list_of_lists(pks) + ["xbrl_factoid"]
+    return pks
+
+
+def get_value_col(tbl_names: list[str]) -> str:
+    """Get the value column for the exploded tables."""
+    value_cols = []
+    for table_name in tbl_names:
+        value_cols.append(
+            pudl.transform.ferc1.FERC1_TFR_CLASSES[
+                table_name
+            ]().params.reconcile_table_calculations.column_to_check
+        )
+    if len(set(value_cols)) != 1:
+        raise ValueError(
+            "Exploding FERC tables requires tables with only one value column. Got: "
+            f"{set(value_cols)}"
+        )
+    value_col = list(set(value_cols))[0]
+    return value_col
+
+
+def remove_factoids_from_mutliple_tables(
```

Comment:
Currently assumes factoids are the same if the _original_ names are the same. After #2623 is done we'll need to revisit this assumption, since we may be comparing the transformed name. We'll need a different way to compare those factoids by name, whether by modifying this method or by writing a different one.